distributed/dashboard/components/scheduler.py

```diff
@@ -0,0 +1,1797 @@
+                        lambda: self.source.stream(new, 10000)
+                    )
+                else:
+                    self.source.stream(new, 10000)
+
+
+class Events(DashboardComponent):
+    def __init__(self, scheduler, name, height=150, **kwargs):
+        self.scheduler = scheduler
+        self.action_ys = dict()
+        self.last = 0
+        self.name = name
+        self.source = ColumnDataSource(
+            {"time": [], "action": [], "hover": [], "y": [], "color": []}
+        )
+
+        x_range = DataRange1d(follow="end", follow_interval=200000)
+
+        fig = figure(
+            title=name,
+            x_axis_type="datetime",
+            height=height,
+            tools="",
+            x_range=x_range,
+            **kwargs
+        )
+
+        fig.circle(
+            source=self.source,
+            x="time",
+            y="y",
+            color="color",
+            size=50,
+            alpha=0.5,
+            legend="action",
+        )
+        fig.yaxis.axis_label = "Action"
+        fig.legend.location = "top_left"
+
+        hover = HoverTool()
+        hover.tooltips = "@action<br>@hover"
+        hover.point_policy = "follow_mouse"
+
+        fig.add_tools(
+            hover,
+            ResetTool(),
+            PanTool(dimensions="width"),
+            WheelZoomTool(dimensions="width"),
+        )
+
+        self.root = fig
+
+    @without_property_validation
+    def update(self):
+        with log_errors():
+            log = self.scheduler.events[self.name]
+            n = self.scheduler.event_counts[self.name] - self.last
+            if log:
+                log = [log[-i] for i in range(1, n + 1)]
+            self.last = self.scheduler.event_counts[self.name]
+
+            if log:
+                actions = []
+                times = []
+                hovers = []
+                ys = []
+                colors = []
+                for msg in log:
+                    times.append(msg["time"] * 1000)
+                    action = msg["action"]
+                    actions.append(action)
+                    try:
+                        ys.append(self.action_ys[action])
+                    except KeyError:
+                        self.action_ys[action] = len(self.action_ys)
+                        ys.append(self.action_ys[action])
+                    colors.append(color_of(action))
+                    hovers.append("TODO")
+
+                new = {
+                    "time": times,
+                    "action": actions,
+                    "hover": hovers,
+                    "y": ys,
+                    "color": colors,
+                }
+
+                if PROFILING:
+                    curdoc().add_next_tick_callback(
+                        lambda: self.source.stream(new, 10000)
+                    )
+                else:
+                    self.source.stream(new, 10000)
+
+
+class SchedulerTaskStream(TaskStream):
```

Comment:
I *think* that there should be only one task stream plot, and that these two classes should be merged.  I'm not entirely sure though.